src/mailchimp/mailchimp_manager.py

```diff
@@ -107,8 +150,40 @@ def upload_certificates(self, attendees: AttendeeManager,
         Raises:
             TODO: Document potential errors.
         """
-        raise NotImplementedError
-
+        #mailchimp.set_config(self.keys) again shouldnt need to set configurations as the set authorisation function handles that
+        operations = []
+        for attendee in attendees:
+            pdf_file = attendee.file_path
+            with open(pdf_file, "rb") as pdf_file:
+                base64_file = base64.b64encode(pdf_file.read())
+            operation = {
+                "method": "POST",
+                "path": f"/file-manager/files",
+                "operation_id": str(attendee.email),  #if there is an attribute to the attendee object for an id or else a name
+                "body":json.dumps({
+                    "name"  : os.path.basename(pdf_file), #returns a string with the base name of the file path
+                    "file_data": base64_file,
+                    "folder_id": folder_id})  
+                    }
+            operations.append(operation)
+
+        payload = {
+                "operations": operations
+                    }
+        response = self._client.batches.start(payload)
+        batch_id = response['id']
+        finished = False
+        while finished == False:
+            response_batch = self._client.batches.status(batch_id)
+            successfull_operations = response_batch['finished_operations']
+            error_operations = response_batch['errored_operations']
+            status = response_batch['status']
+            BatchStatusFunc(status, successfull_operations, error_operations) #updating the callback fucntions to upload with the information of the bath request
+            if status == 'finished':
+                finished = True 
```

Comment:
It would be simpler to use ``status`` as the exit variable. Initialise it to "Pending" before the while loop, then use ``while status != "finished"`` as the exit condition. This way, you don't need the extra ``finished`` variable nor the ``if`` statement.